miden-tx/src/proven_tx/mod.rs

```diff
@@ -123,6 +113,75 @@ impl ProvenTransaction {
+        let known_output_ids = BTreeSet::from_iter(self.output_notes.iter().map(|n| n.note_id()));
+        let unknown_ids: Vec<_> = self
+            .output_note_details
+            .keys()
+            .filter(|&k| !known_output_ids.contains(k))
+            .cloned()
+            .collect();
+        if !unknown_ids.is_empty() {
+            return Err(ProvenTransactionError::NoteDetailsForUnknownNotes(unknown_ids));
+        }
+
+        if !self.account_id.is_on_chain() && self.account_details.is_some() {
+            return Err(ProvenTransactionError::OffChainAccountWithDetails(self.account_id));
+        }
+
+        if self.account_id.is_on_chain() {
+            match self.account_details {
+                None => {
+                    return Err(ProvenTransactionError::OnChainAccountMissingDetails(
+                        self.account_id,
+                    ))
+                },
+                Some(ref details) => {
+                    let is_new_account = self.initial_account_hash == Digest::default();
+
+                    match (is_new_account, details) {
+                        (true, AccountDetails::Delta(_)) => {
+                            return Err(
+                                ProvenTransactionError::NewOnChainAccountRequiresFullDetails(
+                                    self.account_id,
+                                ),
+                            )
+                        },
+                        (true, AccountDetails::Full(account)) => {
+                            if account.id() != self.account_id {
+                                return Err(ProvenTransactionError::AccountIdMismatch(
+                                    self.account_id,
+                                    account.id(),
+                                ));
+                            }
+                            if account.hash() != self.final_account_hash {
+                                return Err(ProvenTransactionError::AccountFinalHashMismatch(
+                                    self.final_account_hash,
+                                    account.hash(),
+                                ));
+                            }
+                        },
+                        (false, AccountDetails::Full(_)) => {
+                            return Err(
+                                ProvenTransactionError::ExistingOnChainAccountRequiresDeltaDetails(
+                                    self.account_id,
+                                ),
+                            )
+                        },
+                        (false, AccountDetails::Delta(_)) => (),
+                    }
+                },
+            }
+        }
+
+        if !skip_proof_checking {
+            let verifier = TransactionVerifier::new(MIN_PROOF_SECURITY_LEVEL);
+            verifier.verify(self).map_err(ProvenTransactionError::ProofVerificationError)?;
+        }
```

Comment:
So - if we want to verify the proof, we need to move `ProvenTransaction` to `miden-tx` crate - right? Are there any other reasons to move besides that?

Reply:
Yes, I had to move `ProvenTransaction` to `miden-tx` because of references to `miden-tx` and `miden-lib`.